src/expr/src/relation.rs

```diff
@@ -302,6 +302,13 @@ pub enum MirRelationExpr {
     },
 }
 
+impl PartialEq for MirRelationExpr {
+    fn eq(&self, other: &Self) -> bool {
+        structured_diff::MreDiff::new(self, other).next().is_none()
```

Comment:
I think it would be good to check this against `Ord` to avoid bugs like the missing join inputs len check. Something like:
```
let result = structured_diff::MreDiff::new(self, other).next().is_none();
soft_assert_eq_no_log!(result, self.cmp(other) == Ordering::Equal);
result
```
(`soft_assert_eq_no_log` doesn't run at all in production, but runs in CI.)

And if we also change `Ord` to a manual implementation in the future, then we could check against equality on a `Debug` print (which is probably a bit more expensive).